### PR TITLE
DTSBPS-462-Controller

### PR DESCRIPTION
**Pull request by gauravchawla73463**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSBPS-462

### Change description ###

Add search by ccd_id to /zip-files endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java [hunk 1]
```diff
@@ -23,8 +23,13 @@ public ZipStatusController(ZipFileStatusService service) {
     }
     // endregion
 
-    @GetMapping
+    @GetMapping(params = {"name"})
     public ZipFileStatus findByFileName(@RequestParam("name") String fileName) {
         return service.getStatusFor(fileName);
     }
+
+    @GetMapping(params = {"ccdId"})
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java:31]: I think param `ccd_id` ?

**Review by mdayican — COMMENTED** [review]
what happens both name and ccdid is provided?

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ZipStatusControllerTest.java [hunk 2]
```diff
@@ -135,6 +137,122 @@ public void should_return_200_with_empty_model_if_no_results_were_found() throws
...
+                emptyList()
+            )
+        );
+
+        List<ZipFileEvent> events = asList(
+            new ZipFileEvent("type0", "container0", now().minusSeconds(10), "reason0"),
+            new ZipFileEvent("type1", "container1", now().minusSeconds(15), "reason1")
+        );
+        String ccdId = "3746374637643";
+        given(service.getStatusByCcdId(ccdId)).willReturn(new ZipFileStatus(null, ccdId, envelopes, events));
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/controller/ZipStatusControllerTest.java:183]: search by ccdid can not return 2 envelopes

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java [hunk 3]
```diff
@@ -24,7 +27,15 @@ public ZipStatusController(ZipFileStatusService service) {
...
+    public ResponseEntity<ZipFileStatus> getStatus(
+        @RequestParam(value = "name", required = false) Optional<String> fileName,
+        @RequestParam(value = "ccd_id", required = false) Optional<String> ccdId) {
+
+        if (fileName.isPresent() && !ccdId.isPresent()) {
+            return ResponseEntity.ok(service.getStatusFor(fileName.get()));
+        } else if (!fileName.isPresent() && ccdId.isPresent()) {
+            return ResponseEntity.ok(service.getStatusByCcdId(ccdId.get()));
+        }
+        return ResponseEntity.badRequest().body(null);
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java:39]: Should be some message.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java [hunk 4]
```diff
@@ -24,7 +27,15 @@ public ZipStatusController(ZipFileStatusService service) {
...
 
     @GetMapping
-    public ZipFileStatus findByFileName(@RequestParam("name") String fileName) {
-        return service.getStatusFor(fileName);
+    public ResponseEntity<ZipFileStatus> getStatus(
+        @RequestParam(value = "name", required = false) Optional<String> fileName,
+        @RequestParam(value = "ccd_id", required = false) Optional<String> ccdId) {
+
+        if (fileName.isPresent() && !ccdId.isPresent()) {
+            return ResponseEntity.ok(service.getStatusFor(fileName.get()));
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java:35]: The service method should be renamed to `getStatusByName` (or `getStatusByFileName`) - similar to `getStatusByCcdId`).

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/controllers/ZipStatusController.java:35]: It has been done in another branch.

**Review by mdayican — APPROVED** [review]

**Review by apereverzin — APPROVED** [review]

**Pull request merged**